### ISSUE TITLE
Annotation masks during training

### ISSUE THREAD

**Issue by Malaikah-Javed** [reginabarzilaygroup/Sybil#63]
@pgmikhael Hi! Did you use those scans which did not have annotations in the json file? Because it raises an error because of the NONE values in the masks. Which consequently does not load the image into the train dataloader.
my train dataloader is also empty for some reason, i think it is because of this.

**Comment by Malaikah-Javed** [comment]
@pgmikhael and how did you use the Non-Annotated images(negative/non-cancerous batch) for training? 
The error for Mask would still occur

**Comment by pgmikhael** [comment]
Hi,

Yes, images with no annotations are still used – the attention loss is just not propagated backwards for those cases. In loading the images, masks with None are made into all-zero arrays.

**Comment by Malaikah-Javed** [comment]
Alright. Thanks!

**Comment by Malaikah-Javed** [comment]
@pgmikhael Hi, hope you are well

How did you perform the calibration on the development set? I know it was using CalibratedClassfierCV(isotonic) from sklearn. What values did you give to make it calibrate from year 1 to year 6 (cumulative probabilities)? How did it fit? Because before calibration the raw risk scores are the same for each year.

There is also the fact that in the NLST dataset, there are only T0, T1, T2 study years. That would make only 3 years of data. 
So how did you manage to make it upto six years of probabilities? It must be in the calibration code but that code is not integrated into the training pipeline. And the calibrator.py file does not tell anything about performing calibration on the development set.

**Comment by pgmikhael** [comment]
Hi,

The calibration is done on each year separately, so the input is the raw score (say 3-year cancer risk) and the target is the corresponding cancer label (cancer diagnosed within 3 years or 3 years of negative followup). 

NLST followed-up participants with surveys to obtain outcomes several years after third screening round (I think up to seven or so years). So, the cancer may not be screen-detectable on the CT, but the future cancer diagnosis is still available.
